src/Lib/Managers/SnapManager.vala

```diff
@@ -0,0 +1,419 @@
+/*
+* Copyright (c) 2019 Alecaddd (https://alecaddd.com)
+*
+* This file is part of Akira.
+*
+* Akira is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+
+* Akira is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+
+* You should have received a copy of the GNU General Public License
+* along with Akira. If not, see <https://www.gnu.org/licenses/>.
+*
+* Authored by: Martin "mbfraga" Fraga <mbfraga@gmail.com>
+*/
+
+public class Akira.Lib.Managers.SnapManager : Object {
+    private const string STROKE_COLOR = "#ff0000";
+    private const double LINE_WIDTH = 2.0;
+    private const double DOT_RADIUS = 3.0;
+    private const double SENSITIVITY = 4.0;
```

Comment:
I wonder if the SENSITIVITY should also account for the canvas_scale.
Maybe if you zoomed in you don't need 4 px but less, and if you're zoomed out you want the items to be more snappy and have higher sensitivity.
Just a thought, we can investigate this later.

Reply:
Yeah, I do think it should be have like that. but that does mean the match calculation needs to be updated on zoom in / out while these operations are happening. But in general how we hook up snapping to the transform operations needs to be thought out a bit. Atm for move you can calculate the snaps from the selection. For other transforms I think only a single snap should be tracked (e.g., which nob is being moved).